### PR TITLE
test: deduplicate remaining tests/unit/core clusters

### LINKED ISSUE #1618: Deduplicate tests/unit/core infra and remaining tests
## Description

The `duplicate-code` CI job (`tools/check_duplicate_code.py`) uses PMD CPD to flag code blocks that appear 3+ times. It currently fails on `main` (`continue-on-error: true` — known baseline). This issue covers eliminating flagged clusters in the core infrastructure and remaining test files not covered by #1616 (service tests) or #1617 (app & lifecycle tests).

Split from #1561 (which covered all of `tests/unit/core/`).

## Scope

- `tests/unit/core/test_block_io_guard.py`
- `tests/unit/core/test_blocking_io_marker_spike.py`
- `tests/unit/core/test_bus_dispatch_semaphore.py`
- `tests/unit/core/test_core_coverage.py`
- `tests/unit/core/test_database_service.py`
- `tests/unit/core/test_event_filter.py`
- `tests/unit/core/test_execution_timeout.py`
- `tests/unit/core/test_log_records.py`
- `tests/unit/core/test_log_records_retention.py`
- `tests/unit/core/test_logging_service.py`
- `tests/unit/core/test_loop_watchdog.py`
- `tests/unit/core/test_main.py`
- `tests/unit/core/test_manifest_repository.py`
- `tests/unit/core/test_migration_runner.py`
- `tests/unit/core/test_param_builders.py`
- `tests/unit/core/test_protect_loop_monkeypatch.py`
- `tests/unit/core/test_runtime_query_service.py`
- `tests/unit/core/test_scheduler_mode_resolution.py`
- `tests/unit/core/test_telemetry_models.py`
- `tests/unit/core/test_telemetry_query_helpers.py`
- `tests/unit/core/test_telemetry_repository.py`
- `tests/unit/core/test_telemetry_repository_errors.py`
- `tests/unit/core/test_unified_execution.py`
- `tests/unit/core/test_web_api_service.py`
- `tests/unit/core/test_web_ui_watcher.py`

Run the tool to see current findings:

```bash
uv run python tools/check_duplicate_code.py 2>&1 | grep -E "tests/unit/core/test_(block_io|blocking_io|bus_dispatch|core_coverage|database_service|event_filter|execution_timeout|log_records|logging_service|loop_watchdog|main|manifest_repository|migration_runner|param_builders|protect_loop|runtime_query|scheduler_mode|telemetry_|unified_execution|web_api_service|web_ui_watcher)"
```

## Approach

For each flagged cluster, choose one of two tracks:

1. **Extract** — pull the repeated pattern into a shared helper, fixture, or parametrized test. This is the default for mechanical boilerplate.
2. **Annotate** — if the duplication is intentional (parallel test structure that is clearer repeated, parametrization that would hurt readability), wrap every occurrence with:
   ```python
   # dup-ignore-start: <reason>
   ...
   # dup-ignore-end
   ```
   All occurrences in a cluster must be annotated — partial annotation still fails. See `tools/check_duplicate_code.py` module docstring for the full annotation mechanism.

## Acceptance Criteria

- [ ] `uv run python tools/check_duplicate_code.py` reports zero un-annotated clusters in the scoped files
- [ ] All existing behavior coverage is preserved — tests may be restructured, but the functionality they verify must remain covered
- [ ] Every `dup-ignore` annotation has a specific reason, not a generic "intentional"

### PR DESCRIPTION
**Pull request by NodeJSmith**
## Summary

Resolves the remaining `tools/check_duplicate_code.py` (PMD CPD) clusters in `tests/unit/core/`
and its cross-directory pairs — the third and final split of #1561, following #1616/#1633
(service tests) and #1617 (app & lifecycle tests). Full investigation, group boundaries, and the
extract-vs-annotate decision framework are documented in
`design/specs/099-dedupe-tests-unit-core/design.md`.

### Loop / blocking-IO group

- `tests/unit/core/test_block_io_guard.py`, `test_loop_watchdog.py`,
  `test_protect_loop_monkeypatch.py`, `test_execution_timeout.py` — extracted repeated
  watchdog/timeout assertion shapes into local helpers/fixtures.
- `tests/integration/test_command_executor.py` — marker-only `dup-ignore` annotation for a
  cluster fragment that lives here; no functional change.

### Telemetry / DB / migration group

- `tests/unit/core/test_telemetry_repository.py`, `test_telemetry_repository_errors.py`,
  `test_manifest_repository.py`, `test_runtime_query_service.py`, `test_migration_runner.py`,
  `test_log_records.py`, `test_log_records_retention.py` — extracted mechanical boilerplate
  shared within the group; deduplication that crossed a real module or test-tier boundary was
  annotated instead of extracted (see design doc's "Extract vs. annotate" section for the two
  confirmed cross-boundary cases).
- `src/hassette/core/database_service.py`, `src/hassette/core/telemetry/repository.py`,
  `src/hassette/web/mappers.py`, `tests/unit/test_schema_migration.py`,
  `tests/unit/test_migration_002.py`, `tests/integration/database/test_database_service_migrations.py`,
  `tests/e2e/mock_fixtures.py`, `tests/unit/test_model_types.py`, `tests/unit/core/conftest.py` —
  marker-only `dup-ignore` annotations; each carries a specific reason rather than a generic one,
  since the paired fragments belong to independent layers (DB-params vs. API-response) or
  independent test tiers that shouldn't be coupled just to satisfy the checker.

### Logging service group

- `tests/unit/core/test_logging_service.py` — extracted repeated setup/assertion patterns.

### Main / web API service group

- `tests/unit/core/test_main.py`, `tests/unit/core/test_web_api_service.py` — extracted repeated
  setup/assertion patterns.

### Unified execution group

- `tests/unit/core/test_unified_execution.py` — extracted repeated setup/assertion patterns
  (largest single-file reduction in the branch).

Closes #1618

### CODE REVIEW

File: tests/unit/core/test_unified_execution.py [hunk 1]
```diff
@@ -7,90 +7,102 @@
...
+    )
+    return base.model_copy(
+        update={
+            "trigger_context_id": trigger_context_id,
+            "trigger_origin": trigger_origin,
+            "trigger_mode": trigger_mode,
+            "retry_count": retry_count,
+            "attempt_number": attempt_number,
+            "args_json": args_json,
+            "kwargs_json": kwargs_json,
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_unified_execution.py]: Fixed — added `build_execution_minimal()`/`build_execution_record_minimal()` that construct `Execution`/`ExecutionRecord` from only their required fields, and switched the default-only tests in both `TestExecutionModelNewColumns` and `TestExecutionRecordNewColumns` to use them. Previously `build_execution()`'s `model_copy(update=...)` and `make_execution_record()`'s explicit kwargs both pinned every new-column field, so a changed or removed model/dataclass default wouldn't have failed these tests. Verified by temporarily flipping `Execution.retry_count`'s default to 1 — the updated test now fails as expected.

File: tests/unit/core/test_unified_execution.py [hunk 2]
```diff
@@ -7,90 +7,132 @@
...
+        duration_ms=duration_ms,
+        status=status,
+        error_type=error_type,
+        error_message=error_message,
+        execution_id=execution_id,
+        listener_id=listener_id,
+        job_id=job_id,
+    )
+    return base.model_copy(
+        update={
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply tests/unit/core/test_unified_execution.py]: Fixed — build_execution() now constructs `Execution(...)` directly with every field instead of building a base via `make_execution()` and layering the rest on with `model_copy(update=...)`, which skips Pydantic validation entirely. Verified by renaming `trigger_context_id` on the model: `test_handler_only_fields_present_on_handler` and `test_handler_only_fields_default_none_on_job` now fail as expected (they silently passed before the fix). The now-unused `make_execution` import was removed.
